ee/src/main/java/org/jboss/as/ee/component/deployers/StartupCountdown.java

```diff
@@ -47,6 +47,22 @@ public void await() throws InterruptedException {
     }
   }
 
+  public boolean tryAwait(long timeout) throws InterruptedException {
+    if (isPrivileged()) return true;
+    if (count != 0) {
+      synchronized (this) {
+        long start = System.currentTimeMillis();
```

Comment:
These constructs should always be using System.nanoTime(), not currentTimeMillis().

Reply:
Thanks, updated to use nanoTime